data/items.ts

```diff
@@ -230,7 +235,13 @@ export const Items: {[itemid: string]: ItemData} = {
 			basePower: 100,
 			type: "Ground",
 		},
-		onUpdate(pokemon) {
+		onStart(pokemon) {
+			if (pokemon.hp <= pokemon.maxhp / 4 || (pokemon.hp <= pokemon.maxhp / 2 && pokemon.hasAbility('gluttony'))) {
+				pokemon.eatItem();
+			}
+		},
+		onTakeDamage(damage, pokemon, source, effect) {
+			if (effect?.id === 'confusion') return;
```

Comment:
This check seems to be completely unnecessary everywhere you put it because of the way you set up the new event.